### ISSUE TITLE
Sqlalchemy error when building jupyterhub image from tags older than 3.1.1

### ISSUE THREAD

**Issue by vladfreeze** [jupyterhub/jupyterhub#4329]
### Bug description

I tried building images based on older jupyterhub tags and the following error pops out (tag 3.1.1 still works):
```Traceback (most recent call last):
      File "/usr/local/lib/python3.10/dist-packages/jupyterhub/app.py", line 3294, in launch_instance_async
        await self.initialize(argv)
      File "/usr/local/lib/python3.10/dist-packages/jupyterhub/app.py", line 2817, in initialize
        self.init_db()
      File "/usr/local/lib/python3.10/dist-packages/jupyterhub/app.py", line 1788, in init_db
        self.session_factory = orm.new_session_factory(
      File "/usr/local/lib/python3.10/dist-packages/jupyterhub/orm.py", line 1063, in new_session_factory
        check_db_revision(engine)
      File "/usr/local/lib/python3.10/dist-packages/jupyterhub/orm.py", line 954, in check_db_revision
        current_table_names = set(inspect(engine).get_table_names())
      File "/usr/local/lib/python3.10/dist-packages/sqlalchemy/inspection.py", line 111, in inspect
        ret = reg(subject)
      File "/usr/local/lib/python3.10/dist-packages/sqlalchemy/engine/reflection.py", line 304, in _engine_insp
        return Inspector._construct(Inspector._init_engine, bind)
      File "/usr/local/lib/python3.10/dist-packages/sqlalchemy/engine/reflection.py", line 237, in _construct
        init(self, bind)
      File "/usr/local/lib/python3.10/dist-packages/sqlalchemy/engine/reflection.py", line 248, in _init_engine
        engine.connect().close()
      File "/usr/local/lib/python3.10/dist-packages/sqlalchemy/engine/base.py", line 3245, in connect
        return self._connection_cls(self)
      File "/usr/local/lib/python3.10/dist-packages/sqlalchemy/engine/base.py", line 174, in __init__
        self.dispatch.engine_connect(self)
      File "/usr/local/lib/python3.10/dist-packages/sqlalchemy/event/attr.py", line 487, in __call__
        fn(*args, **kw)
      File "/usr/local/lib/python3.10/dist-packages/sqlalchemy/event/legacy.py", line 100, in wrap_leg
        return fn(*conv(*args))
      File "/usr/local/lib/python3.10/dist-packages/jupyterhub/orm.py", line 920, in ping_connection
        connection.scalar(select([1]))
      File "/usr/local/lib/python3.10/dist-packages/sqlalchemy/sql/_selectable_constructors.py", line 493, in select
        return Select(*entities)
      File "/usr/local/lib/python3.10/dist-packages/sqlalchemy/sql/selectable.py", line 5219, in __init__
        self._raw_columns = [
      File "/usr/local/lib/python3.10/dist-packages/sqlalchemy/sql/selectable.py", line 5220, in <listcomp>
        coercions.expect(
      File "/usr/local/lib/python3.10/dist-packages/sqlalchemy/sql/coercions.py", line 413, in expect
        resolved = impl._literal_coercion(
      File "/usr/local/lib/python3.10/dist-packages/sqlalchemy/sql/coercions.py", line 652, in _literal_coercion
        self._raise_for_expected(element, argname)
      File "/usr/local/lib/python3.10/dist-packages/sqlalchemy/sql/coercions.py", line 1143, in _raise_for_expected
        return super()._raise_for_expected(
      File "/usr/local/lib/python3.10/dist-packages/sqlalchemy/sql/coercions.py", line 711, in _raise_for_expected
        super()._raise_for_expected(
      File "/usr/local/lib/python3.10/dist-packages/sqlalchemy/sql/coercions.py", line 536, in _raise_for_expected
        raise exc.ArgumentError(msg, code=code) from err
    sqlalchemy.exc.ArgumentError: Column expression, FROM clause, or other columns clause element expected, got [1]. Did you mean to say select(1)?

```
#### Expected behaviour

The normal launching of jupyterhub

#### Actual behaviour

JupyterHub can't run because of the sqlalchemy error.

### How to reproduce

1. Change to a branch older than 3.1.1 (tried with 3.0.0)
2. Run `docker build -t <tagname> .` using the dockerfile from the main `jupyterhub` folder
3. Run the built image: `docker run-t <tagname> `

### Your personal set up

Did a new git clone for testing, selected the 3.0.0 tag and ran the docker commands.

**Comment by minrk** [comment]
Yes, this is fixed in 3.1.1. We can't fix past tags, so the fix is to make a new release, which we did with 3.1.1. If you want to backport a fix, you can do that in your own branch or use your own image-building pipeline. Can I ask why you are rebuilding images with this repo, but not use the images _published_ by this repo?

**Comment by vladfreeze** [comment]
Hello, as an example, I created a branch based on tag 3.0.0 and merged it with my own changes and afterwards planned to keep updating my custom features on this branch, without moving on to a newer jupyterhub version. But in this case I will use the newer versions and keep jupyterhub up to date.
